### PR TITLE
Internal refacto of SSH functions

### PR DESCRIPTION
**Pull request by pyrech**

### CODE REVIEW

File: src/Runner/SshRunner.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        string $path,
+        string $host,
+        string $user,
+        array $sshOptions = [],
+        ?bool $quiet = null,
+        ?bool $allowFailure = null,
+        ?bool $notify = null,
+        ?float $timeout = null,
+    ): Process {
+        $ssh = $this->configuration($host, $user, $sshOptions);
```
Comment (lyrixx) [inline src/Runner/SshRunner.php:27]: this is weird for me to have a var called `ssh` but the method is `configuration`
I would call the method `buildSsh`

File: src/Runner/SshRunner.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    /**
+     * @param array{
+     *     'port'?: int,
+     *     'path_private_key'?: string,
+     *     'jump_host'?: string,
+     *     'multiplexing_control_path'?: string,
+     *     'multiplexing_control_persist'?: string,
+     *     'enable_strict_check'?: bool,
+     *     'password_authentication'?: bool,
+     * } $sshOptions
```
Comment (lyrixx) [inline src/Runner/SshRunner.php:72]: Could be a type in this class https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases

**Review by lyrixx — COMMENTED** [review]
Love it

File: src/Runner/SshRunner.php (hunk 2 above)
Comment (pyrech) [reply src/Runner/SshRunner.php]: Already changed :+1:

File: src/Runner/SshRunner.php (hunk 1 above)
Comment (pyrech) [reply src/Runner/SshRunner.php]: Changed

**Review by lyrixx — APPROVED** [review]

**Comment by lyrixx** [comment]
Thanks

**Pull request merged**